funding/manager.go

```diff
@@ -2779,6 +3004,84 @@ func (f *Manager) annAfterSixConfs(completeChan *channeldb.OpenChannel,
 	return nil
 }
 
+// waitForZeroConfChannel is called when the state is addedToRouterGraph with
+// a zero-conf channel. This will wait for the real confirmation, add the
+// confirmed SCID to the router graph, and then announce after six confs.
+func (f *Manager) waitForZeroConfChannel(c *channeldb.OpenChannel,
+	pendingID [32]byte) error {
+
+	// First we'll check whether the channel is confirmed on-chain. If it
+	// is already confirmed, the chainntnfs subsystem will return with the
+	// confirmed tx. Otherwise, we'll wait here until confirmation occurs.
+	confChan, err := f.waitForFundingWithTimeout(c)
+	if err == ErrConfirmationTimeout {
+		return f.fundingTimeout(c, pendingID)
+	} else if err != nil {
+		return fmt.Errorf("error waiting for zero-conf funding "+
+			"confirmation for ChannelPoint(%v): %v",
+			c.FundingOutpoint, err)
+	}
+
+	// We'll need to refresh the channel state so that things are properly
+	// populated when validating the channel state. Otherwise, a panic may
+	// occur due to inconsistency in the OpenChannel struct.
+	err = c.RefreshShortChanID()
+	if err != nil {
+		return fmt.Errorf("unable to refresh channel state: %v", err)
+	}
+
+	// Now that we have the confirmed transaction and the proper SCID,
+	// we'll call ValidateChannel to ensure the confirmed tx is properly
+	// formatted.
+	err = f.cfg.Wallet.ValidateChannel(c, confChan.fundingTx)
```

Comment:
One outstanding question here is what to do when the tx is double spent. We'll probably need to shutdown the link, clear the channel state, and give up?

Reply:
Yep, we should generally take a look at overhauling how we handle re-org'd channels and also double spent channels for 0.16.